kratos/containers/model.cpp

```diff
@@ -33,27 +34,44 @@ void GetNameWithAscendants(const ModelPart& rModelPart, std::vector<std::string>
         GetNameWithAscendants(rModelPart.GetParentModelPart(), rModelPartNames);
     }
 }
+
 } // empty namespace for internal functions
 
 void Model::Reset()
 {
     mRootModelPartMap.clear();
 }
 
+void Model::CreateRootModelPart(const std::string ModelPartName, ModelPart::IndexType NewBufferSize)
+{
+    auto pvar_list = Kratos::make_intrusive<VariablesList>();
+
+    ModelPart* pmodel_part = new ModelPart(ModelPartName, NewBufferSize, pvar_list, *this );
+    mRootModelPartMap[ModelPartName] = std::unique_ptr<ModelPart>(pmodel_part); // note that i create it separately since Model is friend of ModelPart but unique_ptr is not
```

Comment:
```suggestion
    mRootModelPartMap[ModelPartName] = std::unique_ptr<ModelPart>(p_model_part); // note that i create it separately since Model is friend of ModelPart but unique_ptr is not
```